mmf/configs/defaults.yaml

```diff
@@ -161,7 +161,11 @@ run_type: train_inference
 
 # Configuration for optimizer, examples can be found in models' configs in
 # configs folder
-optimizer: {}
+optimizer:
+    # Whether to allow some of the model's parameters not to be used by the
+    # optimizer. Default is false to guard against missing parameters due to
+    # implementation errors in a model's get_optimizer_parameters
+    allow_unused_parameters: false
```

Comment:
Thoughts on whether this can break any of the existing models?

Reply:
I don't think it breaks our current models, but it may affect probably some custom-implemented user models that implement getting parameters other than get_optimizer_parameters. For example, a generative adversarial network where discriminator parameters are fetched separately and fed into a separate optimizer.

However, users can always set `optimizer.allow_unused_parameters` to True to bypass this check.